pixi/backend/api.js

```diff
@@ -15,17 +15,78 @@
  */
 
 const express = require('express');
+const {lint, LintMode, StatusNumber} = require('@ampproject/toolbox-linter');
+const cheerio = require('cheerio');
 const log = require('@lib/utils/log')('Pixi API');
+const LimitedRemoteFetch = require('@lib/utils/limitedRemoteFetch');
+
+const remoteFetch = new LimitedRemoteFetch({
+  requestHeaders: {
+    'Referer': 'https://amp.dev/page-experience/',
+  },
+});
+
+const execLint = async (url) => {
+  const res = await remoteFetch.fetchHtmlResponse(url);
+  const body = await res.text();
+  const context = {
+    $: cheerio.load(body),
+    headers: {},
+    raw: {
+      headers: res.headers,
+      body,
+    },
+    url,
+    mode: LintMode.Amp,
+  };
+  return lint(context);
+};
+
+const mapLintResult = (result) => {
+  return Object.entries(result).reduce((mappedData, [key, checks]) => {
```

Comment:
We should find a better way to merge result messages directly in linter.

I think it'd prefer to directly pass through the linter results. This way the result data needs to be handled in client (where it needs to be read anyway).

Reply:
Unless we have a new Result format that makes more sense than the old and does not hide any information I would not implement any mapping in the linter.

The middleware is explicitly implemented for pixi, so for me it makes sense to return the best possible format that the frontend can use. Why should we transfer more data than needed, or increase the client JavaScript if we can easily handle it in the backend. The mapping will not eat much server time.

Reply:
My concern was not server time but maintainability. If we change a check result in linter we might have to touch code in backend and client as well. I'd prefer to only update a single point on that case. Is data really a concern here? If yes, it makes sense to pre-filter.

Reply:
Data is not really a concern.
The mapping is not implemented check specific, so we only would have to update the backend if the linter result api changes, or if we want more information in the frontend, but in that case I feel it would be fine to touch frontend and backend.
The main point why I did not like to pass the linter result structure directly is that the status for each check can either be a result, or a list of results. Handling both of these makes the frontend code unnecessarily complex (I fear the frontend code will get quite complex without additional hurdles that we can avoid easily).
 
But thinking about your concern about maintainability, I would at least change the format of the mapped result so that we return an object with a `state` property for each check that can then be extended easily.

Reply:
In that case, let's avoid pre-mature optimization and keep things simple by passing the result through. We can always revisit this later in case it's needed.